codec.go

```diff
@@ -97,6 +143,27 @@ func (c *protoJSONCodec) Unmarshal(binary []byte, message any) error {
 	return options.Unmarshal(binary, protoMessage)
 }
 
+func (c *protoJSONCodec) MarshalStable(message any) ([]byte, error) {
+	// protojson does not offer a "deterministic" field ordering, but fields
+	// are still ordered consistently by their index. However, protojson can
+	// output inconsistent whitespace for some reason, therefore it is
+	// suggested to use a formatter to ensure consistent formatting.
+	// https://github.com/golang/protobuf/issues/1373
+	compactedJSON := new(bytes.Buffer)
+	messageJSON, err := c.Marshal(message)
```

Comment:
Minor nit: can we do `c.Marshal` first, so that we can appropriately size the buffer and potentially skip some slice-growth costs?

Reply:
Honestly, it seems unnecessary to allocate even that: we should have no issues re-using the same buffer. Reading the source code of `json.Compact` I can see that it is safe, since it literally can only write less than the input, and as such could never run over itself.

c2015a9e1e70c7d68604301082b1935bf2bbbc1c

Reply:
Nice, that's even better.